### PR TITLE
fix(46803): unused React import doesn't get removed running organize imports with new react jsx transform

### LINKED ISSUE #46803: unused React import doesn't get removed running organize imports with new react jsx transform
# Bug Report

### 🔎 Search Terms

jsx organize imports

### 🕗 Version & Regression Information

vscode 1.62.2
not sure about what version the built-in typescript has

### ⏯ Playground Link

[Playground Link](https://www.typescriptlang.org/play?jsx=4#code/JYWwDg9gTgLgBAJQKYEMDG8BmUIjgcilQ3wG4BYAKCqQA9JY4ATJTFAVwBt4AKASgC8APgA8TYADdRAenFSgA)

### 💻 Code

```tsx
import React from 'react';

export default ()=><div></div>
```
jsconfig.json
```json
{
  "compilerOptions": {
    "jsx": "react-jsx"
  },
}

```

### 🙁 Actual behavior

unused React doesn't get removed
### 🙂 Expected behavior

it should get removed because it's using the new jsx runtime

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #46803

### CODE REVIEW

File: src/services/organizeImports.ts [hunk 1]
```diff
@@ -162,7 +163,7 @@ namespace ts.OrganizeImports {
 
         function isDeclarationUsed(identifier: Identifier) {
             // The JSX factory symbol is always used if JSX elements are present - even if they are not allowed.
-            return jsxElementsPresent && (identifier.text === jsxNamespace || jsxFragmentFactory && identifier.text === jsxFragmentFactory) ||
+            return jsxElementsPresent && (identifier.text === jsxNamespace || jsxFragmentFactory && identifier.text === jsxFragmentFactory) && !(compilerOptions.jsx === JsxEmit.ReactJSX || compilerOptions.jsx === JsxEmit.ReactJSXDev) ||
```
Comment (andrewbranch) [inline src/services/organizeImports.ts:166]: In `importFixes.ts`, I have a very small utility called `jsxModeNeedsExplicitImport`. Do you want to move that to `utilities.ts` and use it here so in the future we fix more of these things at one time if we need to account for new modes?

Comment (a-tarasyuk) [reply src/services/organizeImports.ts:166]: @andrewbranch Do I need to extend it with ReactJSX and ReactJSXDev?

Comment (andrewbranch) [reply src/services/organizeImports.ts:166]: No, my utility says which ones _need_ an import, which is _only_ React and ReactNative. We do not require an import for Preserve, so it does have different behavior than what you have now.

Comment (a-tarasyuk) [reply src/services/organizeImports.ts]: Oke, I've pushed changes. Thanks

**Review by andrewbranch — APPROVED** [review]
Thanks!

**Comment by andrewbranch** [comment]
There’s a baseline change from organizeImports to update.

**Pull request merged**